File: app/javascript/channels/room_channel.js
var clear_form;

App.room = App.cable.subscriptions.create("RoomChannel", {
  connected: function() {},
  disconnected: function() {},
  received: function(data) {
    return $('#messages').prepend(data['message_html']);
  },
  speak: function(content, data_uri, file_name) {
    this.perform('speak', {
      content: content,
      data_uri: data_uri,
      file_name: file_name
    });
    return clear_form('#test_form');
  }
});

clear_form = function(selector) {
  return $(selector).find(":text, :file").val("");
};

$(document).on('keypress', '#content', function(event) {
  var content, file_name, has_content, has_picture, picture, reader;
  if (event.which === 13) {
    content = $.trim($("#content").val());
    has_content = content.length > 0 ? true : false;
    picture = $('#picture');
    has_picture = picture.get(0).files.length > 0 ? true : false;
    if (has_content || has_picture) {
      if (has_picture) {
        file_name = picture.get(0).files[0].name;
        reader = new FileReader();
        reader.readAsDataURL(picture.get(0).files[0]);
        reader.addEventListener("loadend", function() {
          return App.room.speak(content, reader.result, file_name);
        });
      } else {
        App.room.speak(content);
      }
    }
    return event.preventDefault();
  }
});

File: app/channels/room_channel.rb
class RoomChannel < ApplicationCable::Channel
  def subscribed
    stream_from "room_channel"
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  # app/assets/javascripts/channels/room.coffeeで定義した
  # speak関数によって、下記のspeakアクションが呼ばれます。
  def speak(data)
    message = Message.new(
                    content:          data['content'],
                    picture_data_uri: data['data_uri']
              )
    message.save
  end
end